apps/src/templates/curriculumCatalog/noSectionsToAssignDialogs.jsx

```diff
@@ -0,0 +1,65 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import i18n from '@cdo/locale';
+import Button from '@cdo/apps/templates/Button';
+import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
+import Typography from '@cdo/apps/componentLibrary/typography';
+import style from './no_sections_to_assign_dialog.module.scss';
+
+export const SignInToAssignSectionsDialog = ({onClose}) => (
+  <NoSectionsToAssignBaseDialog
+    headerText={i18n.signInToAssign()}
+    helpText={i18n.signInToAssignHelpText()}
+    onClose={onClose}
+    buttonText={i18n.signInOrCreateAccount()}
+    href={`/users/sign_in?user_return_to=${location.pathname}`}
+  />
+);
+
+SignInToAssignSectionsDialog.propTypes = {
+  onClose: PropTypes.func.isRequired,
+};
+
+const NoSectionsToAssignBaseDialog = ({
+  headerText,
+  helpText,
+  onClose,
+  buttonText,
+  href,
+}) => {
```

Comment:
Could the contents of `NoSectionsToAssignBaseDialog` just be the contents of `SignInToAssignSectionsDialog`? Or what is the reasoning for splitting it up in the same file?

Reply:
I probably shouldn't have been foreseeing into the future, but the next PR uses NoSectionsToAssignBaseDialog to create each of the three possible Assign Dialogues: one for signed out users, one for signed-in students, and one of signed-in teachers without sections
https://github.com/code-dot-org/code-dot-org/blob/8c51e6cc62f35f445692fd3d6667a378d6ad097f/apps/src/templates/curriculumCatalog/noSectionsToAssignDialogs.jsx#L9-L50